Fixes #140 -  CoinHours and calculatedHours conflict.

AntiD2ta opened the pull request:
Fixes #140

Changes:
- Add Ticker CalculatedHours for coin skycoin.
- Update code where GetCoins method is called.
- Update some models to use FormatCoins.

Requires testing
no

AntiD2ta committed:
[core] refs #140 - Add Ticker CalculatedHours.

AntiD2ta committed:
[core] refs #140 - Fix GetCoins methods.

AntiD2ta committed:
[core] refs #140 - Fix SkycoinTransactionInput::GetCoins method.

AntiD2ta committed:
[core] refs# 140 - Update coin/skycoin/params.go with the new Ticker.

AntiD2ta committed:
[core] refs# 140 - Add CalculatedHours to AltcoinMetadata.

AntiD2ta committed:
[util] refs# 140 - Fix bug in FormatCoins related to a zero passed as parameter.

AntiD2ta committed:
[models] refs# 140 - Update Output models for use the new Ticker.

AntiD2ta committed:
[models] refs# 140 - Start to update history and wallet models to use FormatCoins.

olemis commented on src/coin/skycoin/models/coin.go:
```diff
@@ -1,168 +1,169 @@
+}
+
+func (txn *SkycoinPendingTransaction) SupportedAssets() []string {
+	return []string{Sky, CoinHour}
```
No calculated coin hours here ?

olemis committed:
Merge branch 'develop' into stdevAntiD2ta_t140_CoinHours_and_calculatedHours_conflict

mauricio1802 commented on src/coin/skycoin/models/coin.go:
```diff
@@ -1,168 +1,168 @@
-		return uint64(0);
+		return uint64(0)
 	}
 	return txn.Transaction.Transaction.Fee
```
I think that an error should be returned if an invalid ticker is passed to the function

olemis replied:
👍

olemis replied:
error instance can be created in code a this point , until error constant be added later to fix #143

AntiD2ta committed:
[util] - Add a method to know if a string belongs to a list.

AntiD2ta committed:
[core] - Add a return error to Transaction::ComputeFee.

AntiD2ta committed:
[core] refs #140 - Add ticker CalculatedHour to SupportedAssets methods.

AntiD2ta committed:
[core] refs #140 - Fix and update ComputeFee methods.

Fixes:
- Update return type to return (uint64, error).
- Check if the param is a valid ticker in SuportedAssets.

AntiD2ta committed:
[core[ refs #140 - More fixes on GetCoins methods.
